packages/webapp/src/components/Animals/RemoveAnimalsModal/index.tsx

```diff
@@ -0,0 +1,185 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import Drawer from '../../Drawer';
+import Button from '../../Form/Button';
+import ReactSelect from '../../Form/ReactSelect';
+import styles from './styles.module.scss';
+import { Controller, useForm, SubmitHandler } from 'react-hook-form';
+import { useTheme, useMediaQuery } from '@mui/material';
+import clsx from 'clsx';
+import Input from '../../Form/Input';
+import { ReactComponent as WarningIcon } from '../../../assets/images/warning.svg';
+import { ReactComponent as CheckIcon } from '../../../assets/images/check-circle.svg';
+import { useTranslation } from 'react-i18next';
+
+const REASON = 'reason';
+const EXPLANATION = 'explanation';
+
+type FormFields = {
+  [REASON]: string;
+  [EXPLANATION]: string;
+};
+
+type RemoveAnimalsModalProps = {
+  isOpen: boolean;
+  onClose: () => void;
+  onConfirm: SubmitHandler<FormFields>;
+  showSuccessMessage: boolean;
+};
+
+export default function RemoveAnimalsModal(props: RemoveAnimalsModalProps) {
+  const { register, handleSubmit, control, watch } = useForm<FormFields>();
+  const { t } = useTranslation();
+  const theme = useTheme();
+  const isMobile = useMediaQuery(theme.breakpoints.down('sm'));
+
+  const options = [
+    {
+      label: t('REMOVE_ANIMALS.SOLD'),
+      value: 'SOLD',
+    },
+    {
+      label: t('REMOVE_ANIMALS.SLAUGHTERED_FOR_SALE'),
+      value: 'SLAUGHTERED_FOR_SALE',
+    },
+    {
+      label: t('REMOVE_ANIMALS.SLAUGHTERED_FOR_CONSUMPTION'),
+      value: 'SLAUGHTERED_FOR_CONSUMPTION',
+    },
+    {
+      label: t('REMOVE_ANIMALS.NATURAL_DEATH'),
+      value: 'NATURAL_DEATH',
+    },
+    {
+      label: t('REMOVE_ANIMALS.CULLED'),
+      value: 'CULLED',
+    },
+    {
+      label: t('common:OTHER'),
+      value: 'OTHER',
+    },
+    {
+      label: t('REMOVE_ANIMALS.CREATED_IN_ERROR'),
+      value: 'CREATED_IN_ERROR',
+    },
+  ];
+
+  const selectedOption = watch(REASON);
+  const isCreatedInErrorSelected = selectedOption === options[options.length - 1].value;
+
+  // For styling dropdown options
+  const getReactSelectClassNames = (value: string) => {
+    if (!value) return '';
+    return clsx(
+      styles.dropDownOption,
+      value === options[options.length - 1].value
+        ? styles.dropDownOptionRed
+        : styles.dropDownOptionGreen,
+    );
```

Comment:
The `Created in error` option actually (and I'm hoping this isn't too hard to implement with this nifty function here 🤞) has a red hover effect, not green. I think that was not explicitly specified (I just inferred from the mobile-style list), but Loïc has confirmed this, and I believe he will add a new screen showing it.